README.md

```diff
@@ -3,11 +3,150 @@
 This project contains the definition of the fulfillment API.
 
 The API is specified using [Protocol Buffers](https://protobuf.dev) and [gRPC](https://grpc.io) definitions inside the
-[`proto`](proto/fulfillment/v1) directory. For example, the defintion for the `ClusterOrder` object that will be used to
+[`proto`](proto/fulfillment/v1) directory. For example, the definition for the `ClusterOrder` object that will be used to
 request the provisioning of a cluster is inside the
 [`cluster_order_type.proto`](proto/fulfillment/v1/cluster_order_type.proto) file, and the definition of the operations to
 list, get, place and cancel an order are in the
 [`cluster_orders_service.proto`](proto/fulfillment/v1/cluster_orders_service.proto) file.
 
 The server will be implemented using Protocol Buffers and gRPC as well, and it will also support plain HTTP and JSON
-access via the [gRPC-Gateway](https://grpc-ecosystem.github.io/grpc-gateway).
\ No newline at end of file
+access via the [gRPC-Gateway](https://grpc-ecosystem.github.io/grpc-gateway).
+
+## Recommended development setup
+
+This is an opinionated setup, the one used by the person that initially wrote this document. You can find other ways
+to prepare your environment. As long as the resulting code works fine you are good.
+
+### Make sure that you have a version 3.12 or newer of Python available
+
+This is needed because the build tools (the `dev.py` script) of the project are written in Python and uses some recent
+features that are only available in Python 3.12 or newer.
+
+In some Linux distributions (RHEL 9, for example) the name of the Python 3.12 binary may be `python3.12`, but you need
+to make sure that it is named just `python`. To do so you can, for example, create a symbolic link in your `~/bin`
+directory that points to that binary:
```

Comment:
I like to use `uv` for managing python projects. For someone consuming the code, it's as simple as:

```
uv sync
```

And then to run anything:

```
uv run ...script or command...
```

This keeps the python version dependency local to the project (and makes managing dependencies super easy).

I'm fine leaving these instructions here for now, but there are a bunch of manual bits around setting up and activating a python virtual environment that we could simplify.

Reply:
`uv` is new for me, will take a look.